src/main.js

```diff
@@ -74,14 +75,45 @@ let load = loader({
 
   github: {
     requires: ['cfg'],
-    setup: ({cfg}) => {
+    setup: async ({cfg}) => {
       let github = new Github({
         promise: Promise,
       });
-      if (cfg.github.credentials.token) {
-        github.authenticate(cfg.github.credentials);
-      }
-      return github;
+
+      // This object insures that the authentication is delayed until we need it.
+      // Also, the authentication happens not just once in the beginning, but for each request.
+      return {
+        getInstallationGithub: async (inst_id) => {
+          let inteToken = jwt.sign(
+            {
+              iss: cfg.github.credentials.integrationId,
+            },
+            cfg.github.credentials.privatePEM,
+            {
+              algorithm: 'RS256',
+              expiresIn: '1m',
+            }
+          );
+          debug('Authenticating as integration...');
+          try {
+            github.authenticate({type: 'integration', token: inteToken});
+            debug('Authenticated as integration');
+          } catch (e) {
+            debug('Authentication as integration failed!');
+            throw e;
+          }
+          var instaToken = await github.integrations.createInstallationToken({
+            installation_id: inst_id,
+          });
+          debug(`Obtained installation token: ${instaToken.token}`);
```

Comment:
We probably shouldn't log the token!  Logging secret values is generally a bad idea, as it's easy to copy/paste logs and accidentally disclose those secrets.  Instead, log the installationId so we know which installation was involved.